lib/practice_exercises.rb

```diff
@@ -1,13 +1,57 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+  i = 0
+  list.length.times do
+    if list[i] == list[i + 1]
+      list.delete_at(i)
```

Comment:
Because `delete_at` shifts all subsequent elements to the left 1 index, it has a time complexity of O(n).

Since this is inside a loop going n times.  You have a total time complexity of O(n<sup>2</sup>)